concent_api/common/decorators.py

```diff
@@ -85,3 +89,35 @@ def decorated(*args: Any, **kwargs: Any) -> None:
                 return func(*args, **kwargs)
 
         return decorated
+
+
+class retry_on_database_error(object):
+
+    def __init__(self, func: Callable) -> None:
+        self.func = func
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        self._initialize_retry_parameters()
+        while True:
+            try:
+                result = self.func(*args, **kwargs)
+                break
+            except IntegrityError:
+                self._validate_if_max_number_of_tries_exceeded()
+        return result
+
+    def _initialize_retry_parameters(self) -> None:
+        self.max_number_of_retries = settings.MAXIMUM_NUMBER_OF_RETRIES  # pylint: disable=attribute-defined-outside-init
```

Comment:
this should be the part of the `__init__`, as it needs to be defined only once

Reply:
:+1: good catch